rezervo/integrations/fsc/booking.py

```diff
@@ -0,0 +1,229 @@
+import json
+import time
+from datetime import datetime
+from typing import Union
+
+import pytz
+import requests
+from requests import Session
+
+from rezervo.consts import WEEKDAYS
+from rezervo.errors import AuthenticationError, BookingError
+from rezervo.integrations.fsc.auth import authenticate_session, fetch_user_details
+from rezervo.integrations.fsc.schedule import fetch_fsc_schedule
+from rezervo.integrations.fsc.schema import (
+    BookingsResponse,
+    BookingType,
+    rezervo_class_from_fsc_class,
+)
+from rezervo.notify.notify import notify_booking
+from rezervo.schemas.config.config import ConfigValue
+from rezervo.schemas.config.user import Class, IntegrationUser
+from rezervo.schemas.schedule import RezervoClass
+from rezervo.utils.logging_utils import err
+from rezervo.utils.str_utils import format_name_list_to_natural
+
+
+def booking_url(auth_session: Session) -> Union[str, AuthenticationError]:
+    user_details = fetch_user_details(auth_session)
+    if isinstance(auth_session, AuthenticationError):
+        return auth_session
+    return f"https://fsc.no/api/v1/auth/customers/{user_details['id']}/bookings/groupactivities"
+
+
+def find_fsc_class_by_id(
+    integration_user: IntegrationUser, config: ConfigValue, class_id: str
+) -> Union[RezervoClass, None, BookingError, AuthenticationError]:
+    print(f"Searching for class by id: {class_id}")
+    fsc_schedule = fetch_fsc_schedule()
```

Comment:
There is, but it is very limited, since it only returns the class description and image, not the entire class. Therefore, you would have to fetch the schedule anyway. I guess we could first fetch the details, then fetch en schedule for that _day_. What do you think is best?

Reply:
its a no